grinder boards & general changes

Spookerton opened the pull request:
:cl: afirpo & spookerton
rscadd: Reagent grinders and juicers can be constructed with printed circuit boards.
tweak: Reagent grinder UI is updated to be more informative.
tweak: Reagent grinder and juicer skill multiplier is now 50%, 67%, 83%, 100%, 100%.
tweak: Reagent grinders and juicers no longer steal fingers.
tweak: Reagent grinders can use any beaker except vials and bowls as product containers. Juicers can use the same, plus shakers.
tweak: Reagent grinders can accept material stacks from sheet snatchers. Juicers can accept pills from pill bottles.
tweak: Reagent grinders and juicers can grind material objects like coins. Not all material objects have chemical products set up, so this is spotty.
/:cl:

![https://i.imgur.com/gIkKg86.png](https://i.imgur.com/gIkKg86.png)

Reimplemented the guts of reagent grinders while I was fixing the construction requirements. I think that's everything.
closes #29192

afirpo committed:
fix grinder board impl, rewrite grinders

SierraKomodo reviewed (changes requested):
- Grinders don't spawn with beakers already inside them anymore. This leaves the bar with no containers that can be used with their grinder.
- The color of the 'light' icon on the grinder feels oddly inconsistent. With no beaker, its yellow. If I insert a beaker, its green. If I grind something, it turns red and stays red until I eject the beaker and it turns yellow. It then turns green again when I reinsert the same beaker.
	- Similar inconsistency with the reagent grinder

Spookerton committed:
add large beaker to mapped reagent grinders

except the ones that already had a beaker

Spookerton commented:
The bar could use their shaker, but I've rebased this forward and added a large beaker to every mapped reagent grinder that didn't have one on the same tile. Fixed the icon thing, but I think that was actually existing behavior, lol.
